tests/core/test_seeder.py

```diff
@@ -269,6 +281,7 @@ async def test_db_processing(seeder_service: DNSServer) -> None:
 
     # now we check all the combinations once (not a stupid amount of times)
     for use_tcp, target_address, request_type in all_test_combinations:
```

Comment:
this looks a bit controversial, instead of using `pytest` parametrization

Reply:
That's what module scope fixtures or such are for?

Reply:
If there's no updating then maybe a module scoped fixture used by a function scoped fixture that creates a transaction and then makes sure to revert it?  This would help isolate tests just in case while retaining an optimization.

Reply:
> If there's no updating then maybe a module scoped fixture used by a function scoped fixture that creates a transaction and then makes sure to revert it? This would help isolate tests just in case while retaining an optimization.

the one issue with making a module scope fixture is that it will depend on a function scoped fixture, as i don't think it will reuse the same file.

Reply:
i changed it to mostly use cached data but the tests still take about 2 seconds each.